src/containers/withAuthorizedAction/AuthorizationModal/AuthorizationModal.tsx

```diff
@@ -0,0 +1,211 @@
+import React from 'react'
+import { useCallback, useEffect, useMemo, useState } from 'react'
+import { Network } from '@dcl/schemas'
+import {
+  AuthorizationStep,
+  AuthorizationModal as BaseAuthorizationModal
+} from 'decentraland-ui'
+import { t } from '../../../modules/translation/utils'
+import {
+  AuthorizationStepAction,
+  AuthorizationStepStatus,
+  Props
+} from './AuthorizationModal.types'
+import { getStepMessage, getSteps } from './utils'
+
+const LOADING_STATUS = [
+  AuthorizationStepStatus.LOADING_INFO,
+  AuthorizationStepStatus.PROCESSING,
+  AuthorizationStepStatus.WAITING
+]
+
+export function AuthorizationModal({
+  authorization,
+  requiredAllowance,
+  currentAllowance,
+  action,
+  authorizationType,
+  revokeStatus,
+  grantStatus,
+  confirmationStatus,
+  error,
+  confirmationError,
+  network,
+  authorizedContractLabel,
+  onClose,
+  onRevoke,
+  onGrant,
+  onAuthorized,
+  onFetchAuthorizations
+}: Props) {
+  const [currentStep, setCurrentStep] = useState(0)
+  const [loading, setLoading] = useState<AuthorizationStepAction>()
+  const [shouldReauthorize, setShouldReauthorize] = useState(false)
+
+  useEffect(() => {
+    onFetchAuthorizations()
+  }, [onFetchAuthorizations])
+
+  const handleRevokeToken = useCallback(() => {
+    onRevoke()
+    setLoading(AuthorizationStepAction.REVOKE)
+  }, [onRevoke])
+
+  const handleGrantToken = useCallback(() => {
+    onGrant()
+    setLoading(AuthorizationStepAction.GRANT)
+  }, [onGrant])
+
+  const handleAuthorized = useCallback(() => {
+    onAuthorized()
+    setLoading(AuthorizationStepAction.CONFIRM)
+  }, [onAuthorized])
+
+  const steps = useMemo(() => {
+    const authSteps = getSteps({
+      authorizationType,
+      network,
+      requiredAllowance,
+      currentAllowance,
+      authorization,
+      authorizedContractLabel
+    })
+    return [
+      ...authSteps,
+      {
+        title: t('@dapps.authorization_modal.confirm_transaction.title', {
+          action: t(`@dapps.authorization_modal.${action}.action`)
+        }),
+        action: t('@dapps.authorization_modal.confirm_transaction.action'),
```

Comment:
should use here `${action}` too?

```suggestion
        action: t('@dapps.authorization_modal.${action}.action'),
```

Reply:
No the button of the last step is always `Confirm transaction`
<img width="398" alt="image" src="https://github.com/decentraland/decentraland-dapps/assets/11800206/4699e9af-e2dc-49a6-b259-2029c17b5d01">